### DISCUSSION TITLE
What besides p5.js?

### DISCUSSION THREAD

**Discussion by katestange** [numberscope/frontscope#336] [Ideas]
So we've long talked about whether it would be possible to add alternatives to p5.js for visualizer writers.  This would require building onto the framework.  One thing we came across that is of interest is that there's a [p5.js svg project](https://github.com/zenozeng/p5.js-svg).  There's often discussion of things like [Plot](https://observablehq.com/plot/) and [three.js](https://threejs.org/).  This discussion is a place to add links to possibilities and discuss which, if any, are worth pursuing.  And maybe how to make it possible.

**Comment by gwhitney** [comment]
Not judging worthwhileness,  [d3.js](https://d3js.org/) is another commonly-used visualization library. Actually Plot is based on d3, so possibly a common base class for d3/plot would work. And for straight-up "traditional" plots/charts, a possibility is [chartjs](https://www.chartjs.org/). And finally, a rule of thumb is not to create infrastructure for a specific library until someone specifically wants to use that library for a particular visualization they are just about to implement -- if it's just for the future, the suitable libraries for different sorts of visualizers might be very different, and so the effort might be wasted.

**Comment by maxd0328** [comment]
This is good to know, I'll take a look at these when I get the chance. Like I was talking about in the other discussion (#335), as long as the visualizer interface stays very generally and out of the way of the specific rendering implementation, we can really throw anything on top of it to provide different interfaces for visualizer writers to work with. For now it's still just p5, but I also looked into tinkering around with custom APIs to suit different purposes as well. Hopefully this will give me some insight into visualizer writing that I can bring up in the meeting.

**Comment by katestange** [comment]
CindyJS seems like an option:  https://cindyjs.org/gallery/main/

**Comment by gwhitney** [comment]
In the category of "letter-number" software packages ;-), it's conceivable to me that the very cool [g9.js](https://omrelli.ug/g9/gallery/) could be of interest for some interactive visualizations.
